sdk/communication/azure-communication-chat/azure/communication/chat/_models.py

```diff
@@ -56,6 +55,57 @@ def _to_generated(self):
             share_history_time=self.share_history_time
         )
 
+class ChatAttachment(object):
+    """An attachment in a chat message.
+
+    All required parameters must be populated in order to send to Azure.
+
+    :ivar id: Id of the attachment. Required.
+    :vartype id: str
+    :ivar attachment_type: The type of attachment. Required. Known values are: "image" and "file".
+    :vartype attachment_type: str or ~azure.communication.chat.models.AttachmentType
+    :ivar extension: The file extension of the attachment, if available.
+    :vartype extension: str
+    :ivar name: The name of the attachment content.
+    :vartype name: str
+    :ivar url: The URL where the attachment can be downloaded.
+    :vartype url: str
+    :ivar preview_url: The URL where the preview of attachment can be downloaded.
+    :vartype preview_url: str
+    """
+
+    def __init__(
+        self,
+        **kwargs # type: Any
```

Comment:
Please expand this to include all the keyword only parameters. You can use the generated model constructor as an example.
We also need all the parameters documented in a constructor docstring - again you can use the generated model as an example.

Reply:
I did this just for consistency with all other existing class __init__ method, do we need to revise all of them to have explicit __init__ parameters? And by `keyword only parameters` do you mean required parameter only (and have optional ones in **kwargs)?

Reply:
This guidance is based on whether a model should be constructed by a customer or not. If a customer needed to create this object, having a constructor use only `kwargs` is a poor documentation experience. However, as has been clarified, it seems this model will currently be output only, so properly documenting the constructor is not required at this stage.
If we have any models here that are expected to be customer-constructed, we should properly update them to improve the docs/intellisense experience :)

Reply:
I see, appreciate the explanation, will leave it for now since they are all constructed for outputs only. Will revisit this for better clarity if there is need for customer-constructed change in the API.